frontend/src/component/common/Limit/Limit.tsx

```diff
@@ -0,0 +1,144 @@
+    borderRadius: theme.shape.borderRadiusMedium,
+}));
+
+const StyledWarningIcon = styled(WarningIcon)(({ theme }) => ({
+    color: theme.palette.warning.main,
+}));
+
+const StyledErrorIcon = styled(ErrorIcon)(({ theme }) => ({
+    color: theme.palette.warning.main,
+}));
+
+const Header = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    gap: theme.spacing(1),
+    alignItems: 'center',
+    fontWeight: 'bold',
+    borderBottom: `2px solid ${theme.palette.background.application}`,
+    padding: theme.spacing(3, 4),
+    fontSize: theme.fontSizes.mainHeader,
+}));
+
+const Footer = styled(Box)(({ theme }) => ({
+    padding: theme.spacing(3, 4),
+}));
+
+const Main = styled(Box)(({ theme }) => ({
+    borderBottom: `2px solid ${theme.palette.background.application}`,
+    padding: theme.spacing(3, 4),
+}));
+
+const LimitStats = styled(Box)(({ theme }) => ({
+    marginBottom: theme.spacing(2),
+}));
+
+const LimitExplanation = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'space-between',
+    marginTop: theme.spacing(1.5),
+}));
+
+const ExpandableBox = styled(Box)(({ theme }) => ({
+    flex: 1,
+}));
+
+export const Limit: FC<{
+    name: string;
+    shortName?: string;
+    limit: number;
+    currentValue: number;
+    onClose?: () => void;
+}> = ({ name, shortName, limit, currentValue, onClose }) => {
+    const percentageLimit = Math.round((currentValue / limit) * 100);
+    const belowLimit = currentValue < limit;
+    const threshold = 80;
+
+    if (percentageLimit < threshold) {
+        return null;
+    }
+
+    return (
+        <StyledBox>
+            <Header>
+                <ConditionallyRender
+                    condition={belowLimit}
+                    show={<StyledWarningIcon />}
+                    elseShow={<StyledErrorIcon />}
+                />
+
+                <ConditionallyRender
+                    condition={belowLimit}
+                    show={
+                        <ExpandableBox>
+                            You are nearing the limit for {name}
+                        </ExpandableBox>
+                    }
+                    elseShow={
+                        <ExpandableBox>
+                            You reached the limit for {name}
+                        </ExpandableBox>
+                    }
+                />
+
+                <ConditionallyRender
+                    condition={typeof onClose === 'function'}
+                    show={
+                        <Tooltip title='Close' arrow describeChild>
+                            <IconButton onClick={onClose}>
+                                <CloseIcon />
+                            </IconButton>
+                        </Tooltip>
+                    }
+                />
+            </Header>
+            <Main>
+                <LimitStats>
+                    You have added {currentValue} {shortName ?? name}, that is
+                    equivalent to{' '}
```

Comment:
Very nitpicky and more on the UX side of things: but does it really make sense to say "That is equivalent to 100% of the limit" when you're at 100%? It's a small thing, but maybe we could do something like:

> You have added {currentValue} {shortName ?? name}. To add new {shortName ?? name}, you must first remove some of the existing {shortName ?? name}.

But very happy for that not to be in this PR at all 😄

Reply:
BTW, we can also go past the limit if you did it before we introduced this feature so 110% is also possible

Reply:
We can tweak it in the followup PRs when we have at least a few usages of the component

Reply:
Ah, yeah, you're right. That's interesting. So the limit isn't really a limit in that sense. So yeah, let's keep it.